storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -870,14 +870,14 @@ bool xbcloud_put(Object_store *store, const std::string &container,
   xtrabackup_tablespaces.00000000000000000000) is uploaded to cloud storage to
   determine successful xbcloud "put" operation. */
   std::string last_file_prefix = backup_name + "/xtrabackup_tablespaces";
+  auto last_file_size = last_file_prefix.size();
   bool file_found = false;
-
-  if (!object_list.empty()) {
-    auto cur_file = object_list.back();
-    auto last_file_size = last_file_prefix.size();
+  for (auto cur_file : object_list) {
     if (cur_file.size() >= last_file_size &&
-        cur_file.substr(0, last_file_size).compare(last_file_prefix) == 0)
+        cur_file.substr(0, last_file_size).compare(last_file_prefix) == 0) {
```

Comment:
Can we also add a bit of debug here? As a proposition - to really understand what's gone wrong in case of "broken backup"?

Reply:
not sure what more information is required for user. this failure means PXB can't locate xtrabackup_tablespace file to validate backup is upload completely. Added same before exiting.